std/math.d

```diff
@@ -527,14 +527,21 @@ enum real SQRT1_2 =    SQRT2/2;                               /** $(SQRT)$(HALF)
  * = hypot(z.re, z.im).
  */
 Num abs(Num)(Num x) @safe pure nothrow
-if (is(typeof(Num.init >= 0)) && is(typeof(-Num.init)) &&
+if ((is(typeof(Num.init >= 0)) && is(typeof(-Num.init)) ||
+    (is(Num == short) || is(Num == byte))) &&
     !(is(Num* : const(ifloat*)) || is(Num* : const(idouble*))
     || is(Num* : const(ireal*))))
 {
     static if (isFloatingPoint!(Num))
         return fabs(x);
     else
-        return x >= 0 ? x : -x;
+    {
+        // NOTE: The operation is not safe for BigInt on 32 bits systems
+        static if (is(Num == short) || is(Num == byte))
+            return x >= 0 ? x : cast(Num) (x * -1);
```

Comment:
note about performances: the optimizer [turns the multiplication into an asm NEG](https://godbolt.org/g/m8MHzE) so there are in practice no change at all.